python-package/fairgbm/sklearn.py

```diff
@@ -1033,3 +1199,45 @@ def fit(self, X, y,
                    "eval_at : iterable of int, optional (default=(1, 2, 3, 4, 5))\n"
                    f"{' ':12}The evaluation positions of the specified metric.\n"
                    f"{' ':8}{_early_stop}{_after_early_stop}")
+
+
+class FairGBMClassifier(LGBMClassifier):
+
+    def __init__(self, boosting_type='gbdt', num_leaves=31, 
+                 max_depth=-1, learning_rate=0.1, n_estimators=100, 
+                 subsample_for_bin=200000, class_weight=None, 
+                 min_split_gain=0, min_child_weight=0.001, min_child_samples=20, 
+                 subsample=1, subsample_freq=0, colsample_bytree=1, 
+                 reg_alpha=0, reg_lambda=0, random_state=None, 
+                 n_jobs=-1, silent=True, importance_type='split',
+                 multipler_learning_rate=50_000, constraint_type="FPR", 
+                 global_constraint_type="FPR, FNR", global_target_fpr=0.05, 
+                 global_target_fnr=0.40, **kwargs):
+
+        super().__init__(boosting_type, num_leaves, max_depth, 
+                         learning_rate, n_estimators, subsample_for_bin, 
+                         "constrained_cross_entropy", class_weight, min_split_gain, 
+                         min_child_weight, min_child_samples, subsample, 
+                         subsample_freq, colsample_bytree, reg_alpha, 
+                         reg_lambda, random_state, n_jobs, 
+                         silent, importance_type, multipler_learning_rate=multipler_learning_rate,
+                         constraint_type=constraint_type, global_constraint_type=global_constraint_type,
+                         global_target_fpr=global_target_fpr, global_target_fnr=global_target_fnr)
+
+    def fit(self, X, y,
+            constraint_group,
```

Comment:
https://peps.python.org/pep-3102/#:~:text=Keyword-only%20arguments%20are%20not,therefore%20%27required%20keyword%27%20arguments.

constraint_group should be a madatory key-word-only argument

Reply:
Good improvement, will change :+1: